src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8625,14 +8625,6 @@ private bool IsPossibleStatement(bool acceptAccessibilityMods)
                 case SyntaxKind.IdentifierToken:
                     return IsTrueIdentifier();
 
-                // Accessibility modifiers are not legal in a statement,
-                // but a common mistake for local functions. Parse to give a
-                // better error message.
-                case SyntaxKind.PublicKeyword:
-                case SyntaxKind.InternalKeyword:
-                case SyntaxKind.ProtectedKeyword:
-                case SyntaxKind.PrivateKeyword:
-                    return acceptAccessibilityMods;
```

Comment:
Struggling to think of a case where this would be desirable.  Here's the rational for why it's not desirable.  First, an accessibility modifier (or other modifier beyond async/static/ref) is an enormous signal that this is a normal method (since that's how methods have commonly been written since C# v1).  Local functions are the rare case.  

Second, opting to make these local functions instead of methods is objectively worse for normal coding patterns.  Here's why:

If you make these local functions, then `this.FuncName` just breaks since we don't find the method on 'this'.  However, if you make this a method, then `this.FuncName` works, *as does* plain `FuncName` (since it will see the symbol in the containing scope.

This really breaks normal completoin for following methods in the totally reasonable and normal case of a missing close curly.

Reply:
It feels like the proposed error recovery will work well most of the time because most real code puts accessibility modifiers on all members. It's possible someone will try an accessibility mod on a local function and subsequent statements in the method body will parse strangely. But that seems not too bad to me.

Reply:
exactly.

Reply:
I am not convinced. I commonly refactor code to move what was a class-level method into a local function, and haven't yet removed the accessibility modifier. This recovery pattern will likely end up significantly impacting that pattern of refactoring and result in a significantly worse recovery tree. Please add some dedicated parsing tests with the following structure to show the full tree so I can evaluate the recovery:

```cs
class C
{
    public void M1()
    {
        public void BadLocal1()
        {
        }
    }

    public void M2()
    {
    }
}
```